tests/Unit/Evolution/Systems/CurvedScalarWave/Test_Characteristics.cpp

```diff
@@ -229,34 +233,71 @@ SPECTRE_TEST_CASE("Unit.Evolution.Systems.CurvedScalarWave.Characteristics",
   test_characteristics_compute_tags<1>();
   test_characteristics_compute_tags<2>();
   test_characteristics_compute_tags<3>();
+}
+
+namespace {
+template <size_t SpatialDim>
+void check_max_char_speed(const DataVector& used_for_size) noexcept {
+  CAPTURE(SpatialDim);
+  MAKE_GENERATOR(gen);
 
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<1>::apply(
-            {{DataVector{1., -4., 3.4, 2., 5.},
-              DataVector{22., -8., 190., 6., 4.},
-              DataVector{1., -7., 31., 2., 5.},
-              DataVector{7., 8.9, 4., 2., 1.}}}) == 190.);
-  CHECK(CurvedScalarWave::ComputeLargestCharacteristicSpeed<1>::apply(
-            {{DataVector{1., 4., 3., 2., 5.}, DataVector{2., 8., -10., 6., 4.},
-              DataVector{1., 7., 3., -11., 5.},
-              DataVector{7., 3., 4., 2., 1.}}}) == 11.);
+  // Fraction of times the test can randomly fail
```

Comment:
Can you give a description in a C++ comment about what this test is doing? I'm specifically not understanding the "trials," "failure_tolerance," and "check_min/max" parts

Reply:
Added a description of what the test is trying to achieve.